### ISSUE TITLE
Dispatch / subtyping issue on v0.6.3

### ISSUE THREAD

**Issue by andyferris** [JuliaLang/julia#28093]
Here's one that's biting me hard with generic methods on `StaticArray`s where I want to dispatch on the static vector length but no other information. Not sure if this class of problem is known?

MWE:
```
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.3 (2018-05-28 20:20 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-pc-linux-gnu

julia> abstract type StaticArray{S,T,N} <: AbstractArray{T,N}; end

julia> const StaticVector{N,T} = StaticArray{Tuple{N},T,1}
StaticArray{Tuple{N},T,1} where T where N

julia> f(::AbstractVector{<:StaticVector{2}}) = 2
f (generic function with 1 method)

julia> f(::AbstractVector{<:StaticVector{3}}) = 3
ERROR: TypeError: UnionAll: expected Type, got Int64
```

Seems to not be a problem on v0.7.

**Comment by JeffBezanson** [comment]
Dup of #27361 I think; should be fixed in 0.6.4.

**Comment by andyferris** [comment]
Thanks! Glad to hear it.
